core/src/test/java/org/opensearch/sql/analysis/AnalyzerTest.java

```diff
@@ -862,4 +865,82 @@ public void ad_fitRCF_relation() {
         new AD(AstDSL.relation("schema"), argumentMap)
     );
   }
+
+  @Test
+  public void table_function_with_named_parameters() {
+    assertAnalyzeEqual(
+        LogicalPlanDSL.tableFunction(dsl.query_range_function(
+            dsl.namedArgument("query", DSL.literal("http_latency")),
+            dsl.namedArgument("starttime", DSL.literal(12345)),
+            dsl.namedArgument("endtime", DSL.literal(12345)),
+            dsl.namedArgument("step", DSL.literal(14))), table),
+        AstDSL.tableFunction("prometheus.query_range",
+            unresolvedArg("query", stringLiteral("http_latency")),
+            unresolvedArg("starttime", intLiteral(12345)),
+            unresolvedArg("endtime", intLiteral(12345)),
+            unresolvedArg("step", intLiteral(14))));
```

Comment:
Try to reorder parameters in this test to ensure that they are resolved by names, not by position.